lib/HLSL/HLMatrixBitcastLowerPass.cpp

```diff
@@ -193,31 +197,42 @@ void MatrixBitcastLowerPass::lowerMatrix(Instruction *M, Value *A) {
         SmallVector<Value *, 2> idxList(GEP->idx_begin(), GEP->idx_end());
         DXASSERT(idxList.size() == 2,
                  "else not one dim matrix array index to matrix");
-
-        HLMatrixType MatTy = HLMatrixType::cast(EltTy);
-        Value *matSize = Builder.getInt32(MatTy.getNumElements());
-        idxList.back() = Builder.CreateMul(idxList.back(), matSize);
+        unsigned NumElts = HLMatrixType::cast(EltTy).getNumElements();
```

Comment:
I've gotten in a habit where if I have to do `isa<>` and then `cast<>` to instead just do dyn_cast<>? Does that make sense in DXC? In other words, is there a way to instead do something similar to a `dyn_cast` in the conditional like if (HLMatrixType* MatTy = HLMatrixType::dyncast(EltTy))  for line 192.